ovos_bus_client/util/scheduler.py

```diff
@@ -329,18 +331,23 @@ def shutdown(self):
         """
         Stop the running thread.
         """
-        self._stopping.set()
-        # Remove listeners
-        self.bus.remove_all_listeners('mycroft.scheduler.schedule_event')
-        self.bus.remove_all_listeners('mycroft.scheduler.remove_event')
-        self.bus.remove_all_listeners('mycroft.scheduler.update_event')
-        # Wait for thread to finish
-        self.join()
-        # Prune event list in preparation for saving
-        self.clear_repeating()
-        self.clear_empty()
-        # Store all pending scheduled events
-        self.store()
+        try:
+            self._stopping.set()
+            # Remove listeners
+            self.bus.remove_all_listeners('mycroft.scheduler.schedule_event')
+            self.bus.remove_all_listeners('mycroft.scheduler.remove_event')
+            self.bus.remove_all_listeners('mycroft.scheduler.update_event')
+            # Wait for thread to finish
+            self.join(30)
+            # Prune event list in preparation for saving
+            self.clear_repeating()
+            self.clear_empty()
+            # Store all pending scheduled events
+            self.store()
+            self._running.clear()
+        except Exception as e:
+            self._running.clear()
```

Comment:
should we try a self.store here to ensure we dont lose events?

Reply:
excellent point right after i push green XD right in the ego...